citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -494,4 +500,22 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
     public void deleteLocalGroupCitations(String groupId) {
         citationStore.deleteCitationByGroupId(groupId);
     }
+
+    @Override
+    public ICitation addCitationToReferences(ICitation citation, String referenceCitationKey, String reference) throws SelfCitationException {
+        Set<IReference> references = citation.getReferences();
+        if (references == null) {
+            references = new HashSet<>();
```

Comment:
if this is the case, then `references` will never be added to the citation.

Reply:
This condition is added for the citations that don't have any references already added to them. In that case, there's a probability the references might be set to null. It occurs when a person create the citation, instead of importing it.

Reply:
yes, I understand that, but right now the references of the citation object will still be null (you never set the references on the citation, just the local variable)

Reply:
Removing the condition gave me null pointer exception error when I tried to add reference to a dummy citation that I created in zetero. It does not give error when the new dummy citation is created in citesphere, because then it's initialized to empty array.

Reply:
What I'm saying is that this code:
```
Set<IReference> references = citation.getReferences();
if (references == null) {
            references = new HashSet<>();
}
...
 references.add(newReference);
```
will create a list of references that you can add to. However, if `citation.getReferences()` returned null before the if, it will still return null after the if. The following code wont' throw a nullpointer, but the changes you make to `references` won't be stored.

Reply:
Got the logical error here. I have added a line to set references of citation so that the case you are saying won't happen.